src/lib.rs

```diff
@@ -215,6 +215,34 @@ impl<T: Hash, S: BuildHasher> HashRing<T, S> {
     }
 }
 
+pub struct HashRingIterator<T> {
+    ring: Vec<T>,
```

Comment:
Do you think it would be worth using a `std::vec::IntoIter` here instead of a `Vec` so we don't have to clone the underlying `Vec`?

Reply:
That's a good point and a good solution to what I couldn't really figure out. Change is coming shortly

Reply:
I'm trying to find a clean implementation that doesn't clone the underlying Vec but also doesn't expose the private `Node` type. using Map gives a reference which unfortunately means cloning is required. Thoughts?